spec/support/finalize.cr

```diff
@@ -0,0 +1,29 @@
+class State
+  @@count = {} of Symbol => Int64
+
+  def self.inc(key)
+    @@count[key] = @@count.fetch(key, 0i64) + 1
+  end
+
+  def self.count(key)
+    @@count.fetch(key, 0i64)
+  end
+
+  def self.reset
+    @@count.clear
+  end
+end
+
+def assert_finalizes(key)
+  State.reset
+  State.count(key).should eq(0)
+
+  10.times do
+    obj = yield
+    obj.key = key
+  end
+
+  GC.collect
+
+  State.count(key).should be > 0
```

Comment:
Why not exactly 10?

Reply:
`assert_finalizes` is based on `weak_ref_spec` which uses exactly the same expectations, so I was trying to follow the same style. At least one object (the last one) in the series is not freed (because it's referenced by `obj`). Secondarily, some objects require multiple calls to `GC.collect` to be collected. So I guess it's safe to assume if at least one object of given type is collected, the entire type is collectable.

Reply:
As a follow up: the last object in the iteration is not released because it is still referenced by the `obj` variable outside the loop.